### PR TITLE
feat(cli): Add options for the `start` command

### PR DESCRIPTION
**Pull request by lorenzosignoretti**
**Disclaimer**: Unless I missed it, the project does not yet have a `contribute.md` file (or similar) that outlines the guidelines for new contributors / PRs. I used my best judgement in making these changes. Any feedback is welcomed.

### Description
This PR adds the following options for the `start` command:
```
--port, -p      A port number on which to start the application
--hostname, -H  Hostname on which to start the application (default: 0.0.0.0)
--help, -h      Displays this message
```
These can be used e.g. as follows: `deno task start --port=3000`.

The `PORT` and `HOSTNAME` [environment variables](https://deno.land/manual/tools/task_runner#environment-variables) using `deno task` are also supported.

These options have been chosen based (i) on what is supported by the [std http server](https://github.com/denoland/deno_std/blob/main/http/server.ts#L83-L92) used by Fresh and (ii) on what empirically seen on other frameworks (e.g. [`next start`](https://github.com/vercel/next.js/blob/canary/packages/next/cli/next-start.ts)).

### To-do
- [x] Test new feature
- [ ] Update documentation (will create a separate PR once this is reviewed and the feature is approved)

### Additional comments
- Running `demo fmt` on `cli_tests.ts` formatted lines that this PR did not change (10-263). I'm happy to revert these changes if they shouldn't be included in this PR.

### CODE REVIEW

File: src/server/error.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { bold, red } from "./deps.ts";
+
+export function printError(message: string) {
+  console.error(red(`${bold("error")}: ${message}`));
+}
+
+export function error(message: string): never {
+  printError(message);
+  Deno.exit(1);
+}
```
Comment (lorenzosignoretti) [inline src/server/error.ts:10]: This has been duplicated from https://github.com/denoland/fresh/blob/main/src/dev/error.ts

File: src/server/mod.ts [hunk 2]
```diff
@@ -45,15 +46,59 @@ export interface Manifest {
 
 export { ServerContext };
 
-export async function start(
-  routes: Manifest,
-  opts: StartOptions = {},
-) {
+type KeyOrValueOf<T> = keyof T | T[keyof T];
```
Comment (lorenzosignoretti) [inline src/server/mod.ts:49]: Is there a better place to put this ts util?

**Review by lucacasonato — COMMENTED** [review]
Thanks. I'm not sure I really want the `start` function to get into the business of argument parsing. I'll need to ponder this a bit more.

**Comment by lorenzosignoretti** [comment]
> Thanks. I'm not sure I really want the `start` function to get into the business of argument parsing. I'll need to ponder this a bit more.

@lucacasonato Thanks for the feedback! Feel free to close the PR if you don't find it relevant at this stage. I can also create a new issue referencing this PR to make it easy to revisit the topic in the future.

That said, I wanted to share more background on why I proposed these changes. I guess it may not be uncommon during local development to have other projects taking up port 8000. It may be useful to have a quick way to allocate another port temporarily without having to manually edit the options supplied to `start` on `main.ts` (a change which could end up being erroneously committed).

[Here](https://github.com/denoland/fresh/issues/252#issuecomment-1161501418) is a real-life example where it seems one of the proposed options (`hostname`) could have been helpful.

**Comment by lorenzosignoretti** [comment]
Closing this for the time being, will revisit in case this becomes relevant for the team.

**Pull request closed without merging**